morebits.js

```diff
@@ -1134,6 +1135,52 @@ Morebits.array = {
 	}
 };
 
+/**
+ * select2 doesn't natively support highlightening of search hits so we implement that here
+ * @param {jQuery} $select
+ * @param {boolean} [hasOptionGroups=false] - does the select menu have optgroups?
+ * @requires {select2}
+ */
+mw.select2SearchHighlights = function select2SearchHighlights($select, hasOptionGroups) {
```

Comment:
Obviously this should be called `Morebits.select2SearchHighlights`, but when named that way, it _just doesn't work_. The bizarre console error "Morebits.select2SearchHighlights is not a function" appears. Even changing it to `var select2SearchHighlights` gives the same issue.  

But making it `mw.select2SearchHighlights` or even the technically incorrect `Twinkle.select2SearchHighlights` or just `select2SearchHighlights` (w/o `var` or anything) works fine. 

I can't for the life of me figure out what's wrong. It has been a long while since I came across a frustrating bug. Am I missing something obvious?

Reply:
`Morebits.prototype.select2SearchHighlights = function ...`? Though I don't know if makes sense to namespace it under `Morebits.quickForm` or even `Morebits.quickForm.element`, as we do with the other public methods such as `Morebits.quickForm.element.prototype.append()`

Reply:
`Morebits.prototype.select2SearchHighlights` doesn't work (of course, as we aren't making any `Morebits` object).

Naming it `Morebits.quickForm.select2SearchHighlights` also doesn't work. 

I should note that `Morebits.userIsInGroup` and `Morebits.pageNameRegex` are both (public) functions and work fine.

Reply:
Indeed Morebits is a normal object, not an object type or class. I was just guessing :) I don't have an answer for you. `Morebits.foo = function () { console.log('bar') }` for instance works in the JS console. I can try to help debug when I have time, but I probably wouldn't resort to putting this in `mw` where it doesn't belong.

Reply:
> I probably wouldn't resort to putting this in mw where it doesn't belong.

Yeah of course, this is just a makeshift arrangement for now. We could probably put this in twinkle.js file and call it `Twinkle.select2SearchHighlights`, coz that works.

Reply:
The issue could probably be on just my end. It works fine if I paste the whole of morebits.js into the console. But not when I am loading the file from my localhost import (which is how I usually test, it didn't give any problems thus far).